test/python/circuit/test_equivalence.py

```diff
@@ -103,24 +103,42 @@ def test_add_double_entry(self):
         self.assertEqual(entry[1], second_equiv)
 
     def test_set_entry(self):
```

Comment:
Do you think it's worth adding another test case for parallel edges in the equivalence graph? I only mention it because you had a comment about it in the for loop over the incoming edges.

Reply:
Oh yes, thanks, I meant to do that at the time I wrote the comment, but I got distracted between that and writing the tests, so did those some time later.

Reply:
Done in 13c45777df.